### PR TITLE
Monorepo + fixes

### PR DESCRIPTION
**Pull request by kasohrab**
# Motivation
Making to changes to firmware is a hassle when that change relies upon both the mTrain and firmware repo. To reduce that friction, the best design choice is to merge the two together. This pr makes firmware a monorepo (other than fshare).
# Description
Title. Looks like a lot lol. It seems like it worked since the mTrain git history was preserved.
The +fixes part of the title is me branching off arvind's mtrain and firmware branches (so merging this pr will close those as well, see #148). 
# Other fixes

- unified doxygen setup. 
- set up main.cpp to start tracking module meta data again (since we had a hard time getting FreeRTOS trace to work), will probably actually be made useful in a later pr
- delete unused tests
- lots of cleanup

Later prs will also hopefully add more tests and cleanup the codebases a bit more (and make the doxygen cleaner too). Right now the doxygen actually has everything on it which is good enough for me (before large parts of the codebase were not being tracked by doxygen).

Once this is merged, the mTrain repo will be archived.

### CODE REVIEW

**Review by Arvind-Srinivasan — CHANGES_REQUESTED** [review]
I'll come back and look at approving this once we can confirm robots run in shop with the main build.

File: control/include/drivers/AVR910.hpp
Comment (Arvind-Srinivasan) [inline control/include/drivers/AVR910.hpp:43]: Stuff like this should be in our standard robocup.hpp file

File: control/include/drivers/ICM20948.hpp
Comment (Arvind-Srinivasan) [inline control/include/drivers/ICM20948.hpp:6]: Same with these

File: control/include/drivers/ISM43340.hpp
Comment (Arvind-Srinivasan) [inline control/include/drivers/ISM43340.hpp:12]: Move to robocup.hpp the standard lib includes

File: control/include/robocup/modules/MotionControlModule.hpp
Comment (Arvind-Srinivasan) [inline control/include/robocup/modules/MotionControlModule.hpp:6]: This has a weird include formatting

File: control/mtrain/.gitignore
Comment (Arvind-Srinivasan) [inline control/mtrain/.gitignore:1]: This whole file shouldn't exist

File: control/mtrain/API/cpp/CMakeLists.txt
Comment (Arvind-Srinivasan) [inline control/mtrain/API/cpp/CMakeLists.txt:1]: The "cpp" folder should be deprecated with instead just an API_SRC folder

File: control/mtrain/doc/DoxygenLayout.xml
Comment (Arvind-Srinivasan) [inline control/mtrain/doc/DoxygenLayout.xml:1]: Maybe test to see how we can integrate this into the larger robocup firmware doxygen setup

**Review by Arvind-Srinivasan — CHANGES_REQUESTED** [review]
Before we can merge, a) bunch of minor comments, b) cleaning up the file structure, c) the doxygen docs building should work on the whole, and d) the includes should be cleaned up.

First and foremost, every single test from the mtrain level up should work and imo, we should write even more tests to use as regression benchmarks. Would be good to know every API works for example using control. We should also clean up the tests in control level to have only the most recent hardware tests (I.E. get rid of the ones for an IMU we no longer have)

**Comment by kasohrab** [comment]
being done in #157 instead, but the description I will leave here.

**Comment by Arvind-Srinivasan** [comment]
@kasohrab are we reopening this now 156 is working?

**Comment by kasohrab** [comment]
> @kasohrab are we reopening this now #157 is working?

so on this branch radio wouldn't initialize. I'm not sure what of your changes would cause that and it's hard to debug currently. Also, I'd like to start moving to fix scheduling stuff as soon as I can. 

I was thinking to slowly add some of your tweaks over time until something breaks. But if you know what could have caused that then we could reopen this?

**Pull request closed without merging**
